libkbfs/folder_branch_ops.go

```diff
@@ -5447,31 +5448,54 @@ func (fbo *folderBranchOps) backgroundFlusher(betweenFlushes time.Duration) {
 			// so don't bother waiting for a signal, just get right to
 			// the main attraction.
 			doSelect = false
+		} else if fbo.getCachedDirOpsCount(lState) >=
+			fbo.config.BGFlushDirOpBatchSize() {
+			doSelect = false
 		}
 
 		if doSelect {
 			// Wait until we really have a write waiting.
 			forced := false
 			select {
 			case <-fbo.syncNeededChan:
+				if fbo.getCachedDirOpsCount(lState) >=
+					fbo.config.BGFlushDirOpBatchSize() {
+					forced = true
+				}
 			case <-fbo.forceSyncChan:
 				forced = true
 			case <-fbo.shutdownChan:
 				return
 			}
 
 			if !forced {
-				select {
-				case <-ticker.C:
-				case <-fbo.forceSyncChan:
-				case <-fbo.shutdownChan:
-					return
+				timer := time.NewTimer(fbo.config.BGFlushPeriod())
```

Comment:
I did it this way on purpose to make sure the timer started exactly after the first write was signaled.  Also, a timer is use-once, so I think you are suggesting going back to a ticker.  I'd rather keep the timer so that it's measured exactly from the first write signal.  There's not much overhead in this way, since it's only constructed whenever write activity is actually happening.